python/cudf/cudf/utils/utils.py

```diff
@@ -403,3 +404,19 @@ def _all_bools_with_nulls(lhs, rhs, bool_fill_value):
     if result_mask is not None:
         result_col = result_col.set_mask(result_mask.as_mask())
     return result_col
+
+
+class filtered_deprecation(ExitStack):
```

Comment:
There's a slightly cleaner implementation of this kind of thing here: https://github.com/rapidsai/cudf/blob/341e014ed22e7da1e4b8db66a1d7b6fd5fba98e9/python/cudf/cudf/tests/test_categorical.py#L18-L26

I would try to use the `contextlib.contextmanager` decorator and `yield` like in that snippet, instead of implementing a subclass of `ExitStack`.